Client/Frontend/Browser/TabManager.swift

```diff
@@ -1065,6 +1046,111 @@ class TabManager: NSObject {
       value: count
     )
   }
+  
+  // MARK: - Recently Closed
+  
+  /// Function used to add the tab information to Recently Closed when it is removed
+  /// - Parameter tab: The tab which is removed
+  func addTabToRecentlyClosed(_ tab: Tab) {
+    if let savedItem = createRecentlyClosedFromActiveTab(tab) {
+      RecentlyClosed.insert(savedItem)
+    }
+  }
+  
+  /// Function to add all the tabs to recently closed before the list is removef entirely by Close All Tabs
+  func addAllTabsToRecentlyClosed() {
+    var allRecentlyClosed: [SavedRecentlyClosed] = []
+    
+    tabs(withType: .regular).forEach {
```

Comment:
If I close a private tab, does this function get called and add all the regular tabs?

Reply:
No of-course not, this function is called `only` when Close All Tabs action is invoked in long press tab tray button on toolbar. And it is guarded with `!PrivateBrowsingManager.shared.isPrivateBrowsing` where it is called.